contrib/AdaEurope2018/derivation_count.adb

```diff
@@ -43,15 +46,15 @@ procedure Derivation_Count is
    begin
       --  Skip all nodes that are not the derivation of a tagged type, or that
       --  are "with private".
-      if Node.Kind /= LAL.Ada_Derived_Type_Def then
-         return LAL.Into;
+      if Node.Kind /= LALCO.Ada_Derived_Type_Def then
+         return LALCO.Into;
       end if;
 
       Derived := Node.As_Derived_Type_Def;
       if Derived.F_Has_With_Private.P_As_Bool
-        or else not Derived.P_Is_Tagged_Type
+        or else not Derived.As_Base_Type_Decl.P_Is_Tagged_Type
```

Comment:
This conversion will always fail: `Base_Type_Decl` does not derive from `Derived_Type_Def` and conversely. However I think that all `Derived_Type_Def` nodes have a `Type_Decl` parent node, so this should become `Derived.Parent.As_Type_Decl`.

@Roldak could you confirm?

Reply:
I just tried both variants to run, and @pmderodat new variant is working when prev variant

Reply:
Great, thanks!